torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -118,7 +127,27 @@ class SubgraphSlicer {
     if (node->kind() == prim::Constant) {
       return false;
     }
-    return isDifferentiable(node);
+
+    bool requires_grad = !strict_check_;
+    if (strict_check_) {
+      requires_grad = std::any_of(
+          node->outputs().begin(), node->outputs().end(), [](Value* o) {
+            auto tt = o->type()->cast<TensorType>();
+            if (!tt) {
```

Comment:
We should modify Type::requires_grad so that it can return the optional value and rename the current requires_grad to may_require_grad. This logic here is missing for a number of container types, and while I don't think those types appear in the differential ops, this is a bug waiting to happen.